### PR TITLE
Fixed relative path generation for index URLs

### PR DESCRIPTION
**Pull request by pidpawel**
Overview of the bug: if one want's to do something like `{{ .Site.Pages.BySource "index.md" | .UrlTo }}` and the resulting URL will be `index.html`, `Page.Url` will strip the `index.html` part (especially in the top-level directory) and return an empty string (line 128)… which will break on `if path[0] == '/' {` on line 152 as `path` will have no `[0]` element.

Example error: `executing "page" at <.UrlTo>: error calling UrlTo: runtime error: index out of range [0] with length 0`

### CODE REVIEW

**Comment by pidpawel** [comment]
Sorry. Deleted the branch by accident. Reopening.

**Comment by piranha** [comment]
Hmm... what should it be, if not `index.html`? Just `.` maybe?

**Comment by pidpawel** [comment]
Let me rephrase that as I think that we have a slight misunderstanding here :P
Let's start with `{{ .Site.Pages.BySource "index.md" | .UrlTo }}`.
`.UrlTo` does `page.Rel(other.Url())` (line 141)
`other.Url()` in this case will return `""` (empty string) as `url` (on line 127) will be a "top level" `index.html`. Up to this point everything is fine.
Problem starts in `Rel` as it does not support the case of empty URL properly. In the master version it'll crash with `index out of range [0] with length 0`. As only top level `index.html` will return an empty string as it's `Url()` I think it's reasonable to make `Rel` return the relative path to the root in this case.

**Comment by piranha** [comment]
So in the end we don't get an url like `../../index.html`, right? :-)

**Comment by pidpawel** [comment]
Well, we do get an URL like `../../`.

**Comment by piranha** [comment]
👌  cool, thanks!
